File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        // 끝값 기준으로 오름차순 정렬
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>& b) {
            return a[1] < b[1];
        });

        int answer = 0;
        int currEnd = intervals[0][1]; // 첫 번째 구간의 끝값

        for (int i = 1; i < intervals.size(); i++) {
            // 현재 구간이 겹치는 경우
            if (intervals[i][0] < currEnd) {
                answer++; // 구간 제거
            } else {
                // 겹치지 않으면 currEnd를 업데이트
                currEnd = intervals[i][1];
            }
        }

        return answer;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> a[1] - b[1]); 
        int remove = 0; 
        int end = 0; 
        while(end < intervals.length){
            int currEnd = intervals[end][1]; 
            while(end + 1 < intervals.length && intervals[end+1][0] < currEnd){
                remove++; 
                end++; 
            }
            end++; 
        }

        return remove; 
    }
}